ci/Jenkinsfile

```diff
@@ -70,16 +79,51 @@ pipeline {
   }
 
   stages {
-    stage('Download') {
+     stage('Download') {
       steps {
         script {
-          sh "mkdir '${env.WORKSPACE}/tmp/'"
-        }
-        fileOperations([
-          fileDownloadOperation(url:"${params.BUILD}", userName:"${CI_CREDS_USR}", password:"${CI_CREDS_PSW}", targetLocation:"./tmp", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
-        ])
-        script {
-          sh "tar -zxvf '${env.WORKSPACE}/tmp/StatusIm-Desktop.tar.gz' -C '${env.WORKSPACE}/tmp'"
+          /* Creating directory for application build */
+          sh "mkdir -p '${env.WORKSPACE}/tmp/pkg/'"
+          def job_name = "${JOB_NAME}"
+          if(job_name.endsWith("nightly")){
+
+             /* Nightly regression */
+
+            if(params.BUILD_URL != ""){
+              echo "BUILD_URL value was ignored. Please use 'https://ci.infra.status.im/job/status-desktop/job/e2e/job/manual/' to test specific builds."
+            }
+            copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
+          } else if (job_name.endsWith("manual")){
+
+             /* Manual run and PRs trigger for desktop-qa-automation */
+
+            if(params.BUILD_URL == ""){
+              copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
+            } else {
+              withCredentials([usernamePassword(credentialsId: 'status-im-auto', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]){
+                fileOperations([
+                  fileDownloadOperation(url:"${params.BUILD_URL}", userName: "${USERNAME}", password: "${PASSWORD}", targetLocation:"./tmp/pkg/", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
+                ])
+              }
+            }
+          } else {
+
+            /* Manual run and PRs trigger for status-desktop */
+
+            if(params.BUILD_URL == ""){
+              throw new Exception("BUILD_URL is required field.")
+            } else if(params.BUILD_URL.contains("ci.infra.status.im")) {
+              throw new Exception("Please use 'https://ci.infra.status.im/job/status-desktop/job/manual/' to test specific master/release builds.")
+            }
+            withCredentials([usernamePassword(credentialsId: 'status-im-auto', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]){
```

Comment:
Why do we need credentials of any kind at all? Does this really need to be so complex?

Also, why can't we just use a file parameter? Instead of doing all of this silliness:
https://dumbitdude.com/how-to-upload-a-file-in-jenkins-using-file-parameter/

Reply:
fixed